graphs/ag01.qmd

```diff
@@ -0,0 +1,171 @@
+---
+title: AG1
+subtitle: Utilities for All Graphs
+---
+
+------------------------------------------------------------------------
+
+::: panel-tabset
+## Purpose
+
+In this page we collect standard utilities for plotting which can be applied in principle to all graphs.
+Then we don't need to repeat explaining these in each of the other graph pages.
+So it is kind of a cheat sheet for `ggplot2`.
+General introductions to `ggplot2` and a pdf cheatsheet are also recommended and linked at https://ggplot2.tidyverse.org/.
+
+## Setup
+
+We start by creating a graph.
+Note that we load our `ggplot2.utils` NEST package instead of `ggplot2` so we benefit from additional utilities selected from the `ggplot2` extension package ecosystem.
+Note that `ggplot2.utils` automatically loads also `ggplot2`, there we do not need to do that manually.
+
+We also typically apply the `df_explicit_na()` function to the incoming dataset to convert character to factor variables, and code missing values as an explicit factor level, which avoids downstream problems.
+
+Note that you may still run into warning messages after producing some of the graphs if a continuous variable you want to analyze contains NAs.
+To avoid these warning messages, you can use the `drop_na()` function from `tidyr` in the data manipulation step below to remove the rows containing NAs for the specific numeric column (e.g. `drop_na(BMRKR1)` to remove rows where `BMRKR1` is missing).
+
+```{r, message = FALSE}
+library(tern)
+library(tidyr)
+library(dplyr)
+library(ggplot2.utils)
+
+adsl <- scda::synthetic_cdisc_data("rcd_2022_06_27")$adsl %>%
+  df_explicit_na() %>%
+  drop_na(BMRKR1)
+  
+graph <- ggplot(adsl, aes(BMRKR1)) +
+  geom_histogram(aes(y=after_stat(density)), bins = 30) + 
```

Comment:
thanks, great! Can you quickly run `styler` on the code chunks? E.g. here we would want to have a space between the `y` and the `=` etc. - thanks